llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp

```diff
@@ -8141,6 +8138,54 @@ AMDGPUAsmParser::parseSwizzleBitmaskPerm(int64_t &Imm) {
   return true;
 }
 
+bool AMDGPUAsmParser::parseSwizzleFFT(int64_t &Imm) {
+  using namespace llvm::AMDGPU::Swizzle;
+
+  if (!AMDGPU::isGFX9Plus(getSTI())) {
+    Error(getLoc(), "FFT mode swizzle not supported on this GPU");
+    return false;
+  }
+
+  int64_t Swizzle;
+  SMLoc Loc;
+  if (!parseSwizzleOperand(Swizzle, 0, FFT_SWIZZLE_MAX,
+                           "FFT swizzle must be in the interval [0," +
+                               Twine(FFT_SWIZZLE_MAX) + Twine(']'),
+                           Loc))
+    return false;
+
+  Imm = FFT_MODE_ENC | Swizzle;
+  return true;
+}
+
+bool AMDGPUAsmParser::parseSwizzleRotate(int64_t &Imm) {
+  using namespace llvm::AMDGPU::Swizzle;
+
+  if (!AMDGPU::isGFX9Plus(getSTI())) {
```

Comment:
I now tested that these patterns do indeed work on gfx8, and radv will start using them there. So please consider changing this to gfx8+.

Reply:
Based on the ISA doc, these two modes are not supported on GFX8.

Reply:
Given the information in #28975, that looks to be a mistake in the public ISA doc. Also, I confirmed that these instructions pass the subgroupRotate Vulkan-CTS tests on polaris10.

Reply:
> Given the information in #28975, that looks to be a mistake in the public ISA doc. Also, I confirmed that these instructions pass the subgroupRotate Vulkan-CTS tests on polaris10.

@arsenm Your thoughts?

Reply:
@DadSchoorse I believe this is one of those cases that might happen to work, but for whatever reason was never fully verified in GFX8 hardware. So you can use it at your own risk if you want, but the decision not to document it was deliberate, and for the same reason I would be reluctant to support it in the LLVM assembler/disassembler.

Reply:
FWIW, they seem to work on Fiji also.

> was never fully verified in GFX8 hardware

Can you please elaborate what that means exactly?